api/src/main/java/org/pmiops/workbench/api/ClusterController.java

```diff
@@ -128,16 +128,25 @@ public void setUserService(UserService userService) {
 
   @Override
   public ResponseEntity<ClusterListResponse> listClusters(String billingProjectId) {
+    if (billingProjectId == null) {
+      throw new FailedPreconditionException("Must specify billing project");
+    }
+
+    // future-proofing the transition from using free-tier projects to using the billing buffer:
+    // if the free tier project is set, verify that the project has been properly initialized
+    // and ignore the passed-in parameter.
+    // if it is not set, assume we are now using the billing buffer, and we no longer need to check the project.
+
     User user = this.userProvider.get();
-    // TODO as part of Billing Project epic RW-1205: enable status checking for arbitrary project
-    // until then: continue to check the user's free tier project status
-    // because that is the only project we'll pass in here
-    if (user.getFreeTierBillingProjectName() == null ||
-            ! user.getFreeTierBillingProjectName().equals(billingProjectId) ||
-            user.getFreeTierBillingProjectStatusEnum() != BillingProjectStatus.READY) {
-      throw new FailedPreconditionException(
-          "User billing project is not yet initialized, cannot list/create clusters");
+    if (user.getFreeTierBillingProjectName() != null) {
```

Comment:
I would not override the provided billingProjectId in any situation, I would only run the below status check if the incoming billingProjectId == freeTierBillingProjectName.

Here are the migration events going forwards:
t0. All users have a free tier billing project, all clusters are created within each user's project
t1. Enable 1 project == 1 workspace - all current users still have a free tier billing project, any newly created users don't going forward. All clusters should be created in the billing project associated with the relevant workspace, which may or may not be the free tier billing project.
t2. All workspaces migrated to 1 project == 1 workspace. Older users still have freeTierBillingProjectName set, but it isn't used for anything. Clusters now exclusively get created in new per-workspace projects.
t3. Teardown fields / code for freeTierBillingProjectName/Status, now no users have this.

In all cases, we should just respect the incoming billingProject. IF it is a freeTierBillingProject, we can also check the status since we know it.